### PR TITLE
add `fillMissingKeys` parameters to `addItems` method

### LINKED ISSUE #41: Support `fillMissingKeys` in `addItems` method

### PR DESCRIPTION
**Pull request by isabellebouchard**
I added some unit tests for both `addItems` and `checkKeys` (now named `handleMissingKeys` to be more explicit about the expected behaviour). I refactored both methods to handle edge cases properly and sometimes simplify the code. 

Fixes #41

### CODE REVIEW

File: src/helpers/getUniqueKeys.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { SimpleDataItem } from "../types/SimpleData.types"
+
+export default function getUniqueKeys(data: SimpleDataItem[]): string[] {
```
[the remark this answers is not in the document]
Comment (isabellebouchard) [reply src/helpers/getUniqueKeys.ts:3]: `getUniqueKeys` is actually necessary in order to proceed to that check in the constructor. 

Checking that data items have the same keys as the first data item, ie. `Object.keys(data[0]) === Object.keys(data[i])` is not sufficient. You would miss cases like this one
`data = [{patate: 0}, {patate: 0, poil: 1}]`. This is why you need to retrieve unique keys for the entire data array.

Lmk if that make sense to you, and if you have a simpler idea to go around this edge case!

File: src/helpers/handleMissingKeys.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import { SimpleDataItem } from "../types/SimpleData.types.js"
+import isEqual from "lodash.isequal"
+import log from "./log.js"
+import getUniqueKeys from "./getUniqueKeys.js"
+
+export default function handleMissingKeys(
+    data: SimpleDataItem[],
+    fillMissingKeys = true,
+    uniqueKeys?: string[],
```
[the remark this answers is not in the document]
Comment (isabellebouchard) [reply src/helpers/handleMissingKeys.ts:9]: See in `addItems` where I want to handle missing keys in `dataToBeAdded` with keys from `data`.

File: src/helpers/handleMissingKeys.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import { SimpleDataItem } from "../types/SimpleData.types.js"
+import isEqual from "lodash.isequal"
+import log from "./log.js"
+import getUniqueKeys from "./getUniqueKeys.js"
+
+export default function handleMissingKeys(
+    data: SimpleDataItem[],
+    fillMissingKeys = true,
```
Comment (nshiab) [inline src/helpers/handleMissingKeys.ts:8]:
```suggestion
    fillMissingKeys = false,
```

**Review by nshiab — APPROVED** [review]
